libcloudforensics/providers/kubernetes/cluster.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Kubernetes cluster class, starting point for Kubernetes API calls."""
+from typing import Optional, List
+
+from kubernetes import client
+
+from libcloudforensics import logging_utils
+from libcloudforensics.providers.kubernetes import base, workloads, netpol
+
+logging_utils.SetUpLogger(__name__)
+logger = logging_utils.GetLogger(__name__)
+
+class K8sCluster(base.K8sClient):
+  """Class representing a Kubernetes cluster."""
+
+  def __init__(self, api_client: client.ApiClient) -> None:
+    """Creates a K8sCluster object, checking the API client's authorization.
+
+    This constructor calls an authorization check on the api_client, to see
+    whether it is authorized to do all operations on the cluster. The equivalent
+    check for kubectl would be `kubectl auth can-i '*' '*' --all-namespaces`.
+
+    Args:
+      api_client (client.ApiClient): The API client to the Kubernetes cluster.
+    """
+    super().__init__(api_client)
+    self.__AuthorizationCheck()
+
+  def ListPods(self, namespace: Optional[str] = None) -> List[base.K8sPod]:
+    """Lists the pods of this cluster, possibly filtering for a namespace.
+
+    Args:
+      namespace (str): Optional. The namespace in which to list the pods.
+
+    Returns:
+      List[base.K8sPod]: The list of pods for the namespace, or in all
+        namespaces if none is specified.
+    """
+    api = self._Api(client.CoreV1Api)
+
+    # Collect pods
+    if namespace is not None:
```

Comment:
```suggestion
    if namespace:
```

Reply:
I think we resolved this in another PR: I'd like that we call the namespaced API if the caller specifies a namespace via the keyword argument, even if that namespace is the empty string.

Reply:
Yeah I wasn't sure so I marked it just in case. Thanks for double checking